server/app/views/questiontypes/CurrencyQuestionFragment.html

```diff
@@ -0,0 +1,29 @@
+<div th:fragment="currency(question)"
+    th:with="currencyQuestion=${question.createCurrencyQuestion()},
+             questionId=${#strings.randomAlphanumeric(8)},
+             inputId=${#strings.randomAlphanumeric(8)}"
+    th:id="${questionId}">
+    <!-- Title and Help Text -->
+    <div th:replace="~{questiontypes/QuestionBaseFragment :: titleAndHelpTextSingleInput(${question}, ${questionId}, ${inputId})}"></div>
+
+    <!-- Display errors -->
+    <div th:replace="~{questiontypes/QuestionBaseFragment :: validationErrors(${currencyQuestion.getValidationErrors().get(question.getContextualizedPath())})}"></div>
+    
+    <div class="usa-input-group">
+        <div class="usa-input-prefix">$</div>
+        <input th:with="value=${currencyQuestion.getCurrencyValue().isPresent() ?
+                                currencyQuestion.getCurrencyValue().get().prettyPrint() :
+                                currencyQuestion.getFailedUpdates().getOrDefault(currencyQuestion.getCurrencyPath(), '')}"
+            class="usa-input"
+            inputmode="decimal"
+            currency
+            th:id="${inputId}"
+            th:name="${currencyQuestion.getCurrencyPath()}"
+            th:value="${value}"
+            th:aria-describedby="${questionId} + '-description'"
+            th:aria-required="${!question.isOptional()}"
```

Comment:
should this be `currencyQuestion`? It probably doesn't really matter, but just for symmetry?

Reply:
isOptional is only a method on the ApplicantQuestion (question), not the CurrencyQuestion (currencyQuestion). It's a little irritating to have to sprinkle both in various places, but 🤷